examples/webgl_loader_viewer.html

```diff
@@ -0,0 +1,410 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+	<title>three.js webgl - model viewer with OrbitControls</title>
+	<meta charset="utf-8">
+	<meta name="viewport" content="width=device-width, user-scalable=no, minimum-scale=1.0, maximum-scale=1.0">
+	<style>
+	body {
+		font-family: Monospace;
+		background-color: #000;
+		color: #fff;
+		margin: 0px;
+		overflow: hidden;
+	}
+	#info {
+		color: #fff;
+		position: absolute;
+		top: 10px;
+		width: 100%;
+		text-align: center;
+		z-index: 100;
+		display:block;
+	}
+	#info a, .button { color: #f00; font-weight: bold; text-decoration: underline; cursor: pointer }
+	</style>
+</head>
+
+<body>
+	<div id="info">
+		<a href="http://threejs.org" target="_blank">three.js</a> - Model Viewer test with OrbitControls
+		<br>
+		select model:
+		<select id="fileModel" onchange="reloadModel(this.value);">
+			<option value="models/babylon/skull.babylon">models/babylon/skull.babylon</option>
+			<option value="obj/female02/female02.obj">obj/female02/female02.obj</option>
+			<option value="obj/male02/male02.obj">obj/male02/male02.obj</option>
+			<option value="obj/walt/WaltHead.obj">obj/walt/WaltHead.obj</option>
+			<option value="models/awd/simple/simple.awd">models/awd/simple/simple.awd</option>
+			<option value="models/collada/monster/monster.dae">models/collada/monster/monster.dae</option>
+			<option value="models/collada/kawada-hironx.dae">models/collada/kawada-hironx.dae</option>
+			<option value="models/fbx/xsi_man_skinning.fbx">models/fbx/xsi_man_skinning.fbx</option>
+			<option value="models/gltf/duck/glTF/duck.gltf">models/gltf/duck/glTF/duck.gltf</option>
+			<option value="models/json/teapot-claraio.json">models/json/teapot-claraio.json</option>
+			<option value="models/skinned/marine/marine_anims_core.json">models/skinned/marine/marine_anims_core.json</option>
+			<option value="models/assimp/jeep/jeep.assimp.json">models/assimp/jeep/jeep.assimp.json</option>
+			<option value="models/pcd/Zaghetto.pcd">models/pcd/Zaghetto.pcd</option>
+			<option value="models/sea3d/mascot.tjs.sea">models/sea3d/mascot.tjs.sea</option>
+			<option value="models/vrml/house.wrl">models/vrml/house.wrl</option>
```

Comment:
I think the most compatible way would to use an index file of all models in three.js examples/models and examples/obj directories. This index file would need to be added to the git repo, automatically or systematically generated and maintained.

At first, I did think of getting file lists directly from these two directories, but it is not possible because of client side javascript security restriction. I did run across articles explaining that File API were considered for HTML5 specs that would allow to list content of directories and retrieve full local path but it seems to be only supported by Chrome and Opera at the moment: http://caniuse.com/#feat=filesystem

Also, if we used node.js with three.js,  we could use server side javascript and get file lists that way.  So a three.js model viewer version using node.js might be more flexible. It could be part of the npm three.js-node package.

I researched how we could build an index file, and found that with the help the following find command on Mac OS X, we can generate a list that can used to generate an index file: 

find . -type f -print0 | xargs -0 ls -t

For the "examples/models" directory, it currently displays:

./.DS_Store
./collada/.DS_Store
./animated/.DS_Store
./obj/.DS_Store
./skinned/marine/marine_anims_core.blend
./skinned/marine/marine_anims_core.json
./skinned/marine/marine_anims_single.blend
./skinned/marine/marine_anims_single.json
./skinned/marine/marine_ik.blend
./skinned/marine/marine_ik.json
./skinned/simple/simple.blend
./skinned/simple/simple.js
./stl/ascii/pr2_head_pan.stl
./stl/ascii/pr2_head_tilt.stl
./stl/ascii/slotted_disk.stl
./stl/binary/colored.stl
./stl/binary/pr2_head_pan.stl
./stl/binary/pr2_head_tilt.stl
./utf8/James_Body_Lores.jpg
./utf8/James_EyeLashBotTran.png
./utf8/James_EyeLashTopTran.png
./utf8/James_Eye_Green.jpg
./utf8/James_Eye_Inner_Green.jpg
./utf8/James_Face_Color_Hair_Lores.jpg
./utf8/James_Mouth_Gum_Lores.jpg
./utf8/James_Tongue_Lores.jpg
./utf8/MCasShoe1TEX_Lores.jpg
./utf8/MJeans1TEX_Lores.jpg
./utf8/MTshirt3TEX_Lores.jpg
./utf8/Nail_Hand_01_Lores.jpg
./utf8/WaltHi.js
./utf8/WaltHi.utf8
./utf8/ben.js
./utf8/ben.utf8
./utf8/ben_dds.js
./utf8/dds/James_Body_Lores.dds
./utf8/dds/James_EyeLashBotTran.dds
./utf8/dds/James_EyeLashTopTran.dds
./utf8/dds/James_Eye_Green.dds
./utf8/dds/James_Eye_Inner_Green.dds
./utf8/dds/James_Face_Color_Hair_Lores.dds
./utf8/dds/James_Mouth_Gum_Lores.dds
./utf8/dds/James_Tongue_Lores.dds
./utf8/dds/MCasShoe1TEX_Lores.dds
./utf8/dds/MJeans1TEX_Lores.dds
./utf8/dds/MTshirt3TEX_Lores.dds
./utf8/dds/Nail_Hand_01_Lores.dds
./utf8/hand.jpg
./utf8/hand.js
./utf8/hand.utf8
./vrml/house.wrl
./vrml/simple.wrl
./vtk/bunny.vtk
./vtk/cube_ascii.vtp
./vtk/cube_binary.vtp
./vtk/cube_no_compression.vtp
./vtk/liver.vtk
./obj/vive-controller/onepointfive_spec.png
./obj/vive-controller/onepointfive_texture.png
./obj/vive-controller/vr_controller_vive_1_5.obj
./pcd/Zaghetto.pcd
./platform/platform.jpg
./platform/platform.json
./ply/ascii/dolphins.ply
./ply/ascii/dolphins_colored.ply
./ply/binary/Lucy100k.ply
./ply/binary/dolphins_be.ply
./ply/binary/dolphins_le.ply
./sea3d/keyframe.sea
./sea3d/keyframe.tjs.sea
./sea3d/mascot.sea
./sea3d/mascot.tjs.sea
./sea3d/morph.sea
./sea3d/morph.tjs.sea
./sea3d/robot.o3dgc.tjs.sea
./sea3d/robot.sea
./sea3d/robot.tjs.sea
./sea3d/skin.sea
./sea3d/skin.tjs.sea
./sea3d/sound.sea
./sea3d/sound.tjs.sea
./skin_and_morph.dae
./skinned/UCS/skins/Asian_Male.jpg
./skinned/UCS/skins/Black_Female.jpg
./skinned/UCS/skins/Caucasion_Female.jpg
./skinned/UCS/skins/Caucasion_Male.jpg
./skinned/UCS/skins/Highlighted_Muscles.jpg
./skinned/UCS/skins/Indian_Male.jpg
./skinned/UCS/umich_ucs.js
./skinned/UCS_config.json
./skinned/knight.js
./skinned/marine/M4.png
./skinned/marine/MarineCv2_color.jpg
./skinned/marine/m4.blend
./skinned/marine/m4.js
./skinned/marine/marine_anims.json
./skinned/marine/marine_anims_all.blend
./skinned/marine/marine_anims_all.json
./3mf/cube_gears.3mf
./amf/rook.amf
./animated/flamingo.js
./animated/horse.js
./animated/monster/monster.blend
./animated/monster/monster.jpg
./animated/monster/monster.js
./animated/parrot.js
./animated/sittingBox.js
./animated/stork.js
./assimp/interior/image1.jpg
./assimp/interior/image2.jpg
./assimp/interior/interior.3ds
./assimp/interior/interior.assimp.json
./assimp/interior/wall.jpg
./assimp/jeep/jeep.assimp.json
./assimp/jeep/jeep1.jpg
./assimp/jeep/jeep1.ms3d
./assimp/jeep/jeep1.readme.txt
./awd/simple/simple.awd
./awd/simple/textures/texture.jpg
./babylon/skull.babylon
./bvh/pirouette.bvh
./collada/VWS_B_Male2-2.jpg
./collada/avatar.dae
./collada/kawada-hironx.dae
./collada/monster/monster.dae
./collada/monster/monster.jpg
./collada/monster/readme.txt
./collada/multimaterial.dae
./collada/pump/pump.dae
./collada/pump/pump_body.jpg
./collada/pump/pump_gears.jpg
./collada/pump/pump_metalreflect.jpg
./ctm/LeePerry.ctm
./ctm/WaltHead.ctm
./ctm/ben.ctm
./ctm/camaro/camaro.ctm
./ctm/camaro/camaro.js
./ctm/camaro/car-ao.png
./ctm/camaro/plane-ao-256.png
./ctm/hand.ctm
./fbx/Char_UV_Texture.gif
./fbx/xsi_man.fbx
./fbx/xsi_man_skinning.fbx
./gltf/CesiumMan/README.txt
./gltf/CesiumMan/glTF-MaterialsCommon/Cesium_Man.bin
./gltf/CesiumMan/glTF-MaterialsCommon/Cesium_Man.gltf
./gltf/CesiumMan/glTF-MaterialsCommon/Cesium_Man.jpg
./gltf/CesiumMan/glTF/Cesium_Man.bin
./gltf/CesiumMan/glTF/Cesium_Man.gltf
./gltf/CesiumMan/glTF/Cesium_Man.jpg
./gltf/CesiumMan/glTF/Cesium_Man0FS.glsl
./gltf/CesiumMan/glTF/Cesium_Man0VS.glsl
./gltf/CesiumMilkTruck/README.txt
./gltf/CesiumMilkTruck/glTF-MaterialsCommon/CesiumMilkTruck.bin
./gltf/CesiumMilkTruck/glTF-MaterialsCommon/CesiumMilkTruck.gltf
./gltf/CesiumMilkTruck/glTF-MaterialsCommon/CesiumMilkTruck.png
./gltf/CesiumMilkTruck/glTF/CesiumMilkTruck.bin
./gltf/CesiumMilkTruck/glTF/CesiumMilkTruck.gltf
./gltf/CesiumMilkTruck/glTF/CesiumMilkTruck.png
./gltf/CesiumMilkTruck/glTF/CesiumMilkTruck0FS.glsl
./gltf/CesiumMilkTruck/glTF/CesiumMilkTruck0VS.glsl
./gltf/CesiumMilkTruck/glTF/CesiumMilkTruck1FS.glsl
./gltf/CesiumMilkTruck/glTF/CesiumMilkTruck1VS.glsl
./gltf/duck/README.txt
./gltf/duck/glTF-MaterialsCommon/duck.bin
./gltf/duck/glTF-MaterialsCommon/duck.gltf
./gltf/duck/glTF-MaterialsCommon/duckCM.png
./gltf/duck/glTF/duck.bin
./gltf/duck/glTF/duck.gltf
./gltf/duck/glTF/duck0FS.glsl
./gltf/duck/glTF/duck0VS.glsl
./gltf/duck/glTF/duckCM.png
./gltf/monster/glTF-MaterialsCommon/monster.bin
./gltf/monster/glTF-MaterialsCommon/monster.gltf
./gltf/monster/glTF-MaterialsCommon/monster.jpg
./gltf/monster/glTF/monster.bin
./gltf/monster/glTF/monster.gltf
./gltf/monster/glTF/monster.jpg
./gltf/monster/glTF/monster0FS.glsl
./gltf/monster/glTF/monster0VS.glsl
./gltf/monster/readme.txt
./json/pressure.json
./json/scene-animation.json
./json/teapot-claraio.json
./md2/ogro/Ogro.txt
./md2/ogro/ogro.md2
./md2/ogro/skins/arboshak.png
./md2/ogro/skins/ctf_b.png
./md2/ogro/skins/ctf_r.png
./md2/ogro/skins/darkam.png
./md2/ogro/skins/freedom.png
./md2/ogro/skins/gib.png
./md2/ogro/skins/gordogh.png
./md2/ogro/skins/grok.jpg
./md2/ogro/skins/igdosh.png
./md2/ogro/skins/khorne.png
./md2/ogro/skins/nabogro.png
./md2/ogro/skins/ogrobase.png
./md2/ogro/skins/sharokh.png
./md2/ogro/skins/weapon.jpg
./md2/ogro/weapon.md2
./md2/ratamahatta/ratamahatta.md2
./md2/ratamahatta/skins/ctf_b.png
./md2/ratamahatta/skins/ctf_r.png
./md2/ratamahatta/skins/dead.png
./md2/ratamahatta/skins/gearwhore.png
./md2/ratamahatta/skins/ratamahatta.png
./md2/ratamahatta/skins/w_bfg.png
./md2/ratamahatta/skins/w_blaster.png
./md2/ratamahatta/skins/w_chaingun.png
./md2/ratamahatta/skins/w_glauncher.png
./md2/ratamahatta/skins/w_hyperblaster.png
./md2/ratamahatta/skins/w_machinegun.png
./md2/ratamahatta/skins/w_railgun.png
./md2/ratamahatta/skins/w_rlauncher.png
./md2/ratamahatta/skins/w_shotgun.png
./md2/ratamahatta/skins/w_sshotgun.png
./md2/ratamahatta/skins/weapon.png
./md2/ratamahatta/w_bfg.md2
./md2/ratamahatta/w_blaster.md2
./md2/ratamahatta/w_chaingun.md2
./md2/ratamahatta/w_glauncher.md2
./md2/ratamahatta/w_hyperblaster.md2
./md2/ratamahatta/w_machinegun.md2
./md2/ratamahatta/w_railgun.md2
./md2/ratamahatta/w_rlauncher.md2
./md2/ratamahatta/w_shotgun.md2
./md2/ratamahatta/w_sshotgun.md2
./md2/ratamahatta/weapon.md2
./mmd/LICENSE
./mmd/Readme.txt
./mmd/audios/readme.txt
./mmd/audios/wavefile_short.mp3
./mmd/miku/eyeM2.bmp
./mmd/miku/miku_v2.pmd
./mmd/miku/readme.txt
./mmd/miku/readme_miku_v2.txt
./mmd/vmds/readme.txt
./mmd/vmds/readme_wavefile.txt
./mmd/vmds/readme_wavefile_camera.txt
./mmd/vmds/wavefile_camera.vmd
./mmd/vmds/wavefile_v2.vmd
./mmd/vpds/01.vpd
./mmd/vpds/02.vpd
./mmd/vpds/03.vpd
./mmd/vpds/04.vpd
./mmd/vpds/05.vpd
./mmd/vpds/06.vpd
./mmd/vpds/07.vpd
./mmd/vpds/08.vpd
./mmd/vpds/09.vpd
./mmd/vpds/10.vpd
./mmd/vpds/11.vpd
./mmd/vpds/readme.txt
./molecules/.htaccess
./molecules/Al2O3.pdb
./molecules/aspirin.pdb
./molecules/buckyball.pdb
./molecules/caf2.pdb
./molecules/caffeine.pdb
./molecules/cholesterol.pdb
./molecules/cocaine.pdb
./molecules/cu.pdb
./molecules/cubane.pdb
./molecules/diamond.pdb
./molecules/ethanol.pdb
./molecules/glucose.pdb
./molecules/graphite.pdb
./molecules/lsd.pdb
./molecules/lycopene.pdb
./molecules/nacl.pdb
./molecules/nicotine.pdb
./molecules/ybco.pdb
./nrrd/I.nrrd
./obj/cerberus/Cerberus.obj
./obj/cerberus/Cerberus_A.jpg
./obj/cerberus/Cerberus_M.jpg
./obj/cerberus/Cerberus_N.jpg
./obj/cerberus/Cerberus_R.jpg
./obj/emerald.obj
./obj/tree.obj

For the "examples/obj" directory, it currently displays:

./.DS_Store
./walt/WaltHead_bin.bin
./walt/WaltHead_bin.js
./walt/WaltHead_slim.js
./Bird.js
./Qrcode.js
./Suzanne.js
./WaltHeadLo.js
./camaro/.htaccess
./camaro/CamaroNoUv_bin.bin
./camaro/CamaroNoUv_bin.js
./cubecolors/.htaccess
./cubecolors/cube_fvc.blend
./cubecolors/cube_fvc.js
./cubecolors/cubecolors.blend
./cubecolors/cubecolors.js
./f50/.htaccess
./f50/F50NoUv_bin.bin
./f50/F50NoUv_bin.js
./female02/.htaccess
./female02/01_-_Default1noCulling.JPG
./female02/02_-_Default1noCulling.JPG
./female02/03_-_Default1noCulling.JPG
./female02/Female02_bin.bin
./female02/Female02_bin.js
./female02/Female02_slim.js
./female02/female02.mtl
./female02/female02.obj
./female02/readme.txt
./gallardo/.htaccess
./gallardo/GallardoNoUv_bin.bin
./gallardo/GallardoNoUv_bin.js
./gallardo/parts/gallardo_body_bin.bin
./gallardo/parts/gallardo_body_bin.js
./gallardo/parts/gallardo_wheel_bin.bin
./gallardo/parts/gallardo_wheel_bin.js
./leeperrysmith/.htaccess
./leeperrysmith/Infinite-Level_02_Disp_NoSmoothUV-4096.jpg
./leeperrysmith/Infinite-Level_02_Tangent_SmoothUV.jpg
./leeperrysmith/LeePerrySmith.js
./leeperrysmith/LeePerrySmith.obj
./leeperrysmith/LeePerrySmith_License.txt
./leeperrysmith/Map-COL.jpg
./leeperrysmith/Map-SPEC.jpg
./lightmap/lightmap-ao-shadow.png
./lightmap/lightmap.js
./lightmap/readme.txt
./lightmap/rocks.jpg
./lightmap/stone.jpg
./lucy/.htaccess
./male02/.htaccess
./male02/01_-_Default1noCulling.JPG
./male02/01_-_Default1noCulling.dds
./male02/Male02_bin.bin
./male02/Male02_bin.js
./male02/Male02_dds.js
./male02/Male02_slim.js
./male02/male-02-1noCulling.JPG
./male02/male-02-1noCulling.dds
./male02/male02.mtl
./male02/male02.obj
./male02/male02_dds.mtl
./male02/orig_02_-_Defaul1noCulling.JPG
./male02/orig_02_-_Defaul1noCulling.dds
./male02/readme.txt
./ninja/NinjaLo_bin.bin
./ninja/NinjaLo_bin.js
./terrain.js
./veyron/.htaccess
./veyron/VeyronNoUv_bin.bin
./veyron/VeyronNoUv_bin.js
./veyron/parts/veyron_body_bin.bin
./veyron/parts/veyron_body_bin.js
./veyron/parts/veyron_wheel_bin.bin
./veyron/parts/veyron_wheel_bin.js
./walt/.htaccess
./walt/WaltHead.mtl
./walt/WaltHead.obj

Maybe we could use a (ro)bot to generate this index file with this find or a similar command available for git/github repo each time. And then we would have a javascript function pull only what it needs from the index file to display the full list of available models.